### PR TITLE
Better async authentication check

### PR DESCRIPTION
**Pull request by xbill82**

### CODE REVIEW

File: public/javascripts/common/authentication/authentication.service.js [hunk 1]
```diff
@@ -5,71 +5,119 @@ angular.module('kuzzle.authentication')
...
+                message: 'Found invalid token in cookie. Not authenticated.'});
+              return;
+            }
+
+            kuzzle.setJwtToken(Session.session.jwtToken);
+            onLoginSuccess(Session.session.jwtToken);
+            deferred.resolve(true);
+          });
+        } else {
+          deferred.reject({
```
Comment (stafyniaksacha) [inline public/javascripts/common/authentication/authentication.service.js]: Should reject with Error object

File: public/javascripts/common/authentication/session.service.js [hunk 2]
```diff
@@ -3,39 +3,55 @@ angular.module('kuzzle.authentication')
...
-    this.session.id = null;
+    this.session.jwtToken = null;
     this.session.userId = null;
-    this.session.userRole = null;
+    this.session.userProfile = null;
 
     $cookies.remove(COOKIE_KEY);
   };
 
+  this.saveToCookie = function () {
```
Comment (stafyniaksacha) [inline public/javascripts/common/authentication/session.service.js]: Can be renamed to 'persist', 
If we change the persistance layer, we will not have to rename it too.

Comment (xbill82) [inline public/javascripts/common/authentication/session.service.js]: yep, definitely!

File: public/javascripts/common/authentication/authentication.service.js (hunk 1 above)
Comment (xbill82) [inline public/javascripts/common/authentication/authentication.service.js]: After a quick discussion we decided that this is not an error, but rather a negative response. Nothing to change here.

**Comment by AnthonySendra** [comment]
+1

**Comment by dbengsch** [comment]
+1

**Pull request merged**